docs/ci.md

```diff
@@ -54,6 +54,92 @@ Received 0 of 614674276 (0.0%), 0.0 MBs/sec
 ...
 ```
 
+### Container-based tests (Testcontainers)
+
+We make heavy use of container-based tests when running integration tests in Zeebe, primarily by leveraging [Testcontainers](https://testcontainers.org).
+
+> This applies only to jobs which run integration tests, such as the update test, or the integration test jobs.
+
+When you run the tests locally, they will use your local Docker environment. However, as part of our CI pipeline, we make use of their SaaS offering, Testcontainers Cloud (TCC). This allows us to decouple resource usage used by containers and provides a modest performance boost to our tests by running the containers on different machines than the ones where our tests run.
+
+#### How it works
+
+Grossly simplified, TCC works by setting up an agent which will act as Docker daemon, but proxy requests to remote VMs. It will then write a [Testcontainers configuration file](https://www.testcontainers.org/features/configuration/) that the Testcontainers library will automatically use when a test is ran.
+
+> Note that AtomicJar specifically mentions that TCC is not a Docker-as-a-Service product, and shouldn't be treated as such.
+
+Make sure to join the private Slack channel Camunda has with AtomicJar around TCC. There you will find a private link to our dashboard which contains documentation around TCC, and can provide feedback, ask qeustions, and receive support.
+
+> If you are not yet in the channel, anybody from the Zeebe team can invite you.
+
+#### Usage
+
+To use TCC, we first need to define a few environment variables to your job.
+
+```yaml
+env:
+  TC_CLOUD_LOGS_VERBOSE: true
+  TC_CLOUD_CONCURRENCY: 4
+  ZEEBE_TEST_DOCKER_IMAGE: localhost:5000/camunda/zeebe:current-test
+```
+
+Let's break this down a bit.
+
+1. Ensure the agent logs are verbose. TCC is still a beta product, so we want to make sure we can provide as much feedback as possible when troubleshooting any issues.
+1. Define how many concurrent remote VMs our tests can use to start containers on.
+1. Specify the name of the image our container based tests should use.
+
+You'll notice the image name starts with a URL, because we'll be using a local Docker registry. This is required because we want to use the image that we build as part of the pipeline, without pushing it to any public Docker registry. By using a local registry and exposing it to TCC, we can use "private" images.
```

Comment:
They suggest it in their docs :smile: An alternative would be using the infra provided Harbor registry, but I don't know if it really brings us much here since this solution isn't really much more complex :shrug: